pilot/pkg/networking/core/v1alpha3/tracing.go

```diff
@@ -350,39 +350,59 @@ func defaultTags() []*tracing.CustomTag {
 		dryRunPolicyTraceTag("istio.authorization.dry_run.allow_policy.result", authz_model.RBACShadowRulesAllowStatPrefix+authz_model.RBACShadowEngineResult),
 		dryRunPolicyTraceTag("istio.authorization.dry_run.deny_policy.name", authz_model.RBACShadowRulesDenyStatPrefix+authz_model.RBACShadowEffectivePolicyID),
 		dryRunPolicyTraceTag("istio.authorization.dry_run.deny_policy.result", authz_model.RBACShadowRulesDenyStatPrefix+authz_model.RBACShadowEngineResult),
+	}
+}
+
+func buildCustomTracingTags(metadata *model.NodeMetadata) []*tracing.CustomTag {
+	var revision, service string
+	if metadata.Labels != nil {
+		revision = metadata.Labels["service.istio.io/canonical-revision"]
+		service = metadata.Labels["service.istio.io/canonical-name"]
+	}
+	if revision == "" {
+		revision = "latest"
+	}
+	if service == "" {
+		service = "unknown"
```

Comment:
this makes me a bit uneasy. it is probably overkill to rerun the fallback for other labels though.

Reply:
This is preserving the original behavior. What would you prefer here (we have more options now)?

Reply:
I think this is actually fine. I was just thinking out loud - sorry about that. 

Maybe we can add a comment here to say "this should have been properly handled during proxy injection." ?

Reply:
Maybe file an issue with the concern and I'll link it here?

Reply:
Added a comment.